### PR TITLE
Added few ras tools tests

### PR DESCRIPTION
**Pull request by shirishaganta1**
Added nvsetenv,usysattn and usysfault test cases

Signed-off-by: Shirisha Ganta <SHIRISHA.Ganta1@ibm.com>

### CODE REVIEW

**Comment by shirishaganta1** [comment]
sles
\# avocado run ras_tools.py 
JOB ID     : 1d0c1dbd188a50045d7238d2afce82866d78840e
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-08-04T16.42-1d0c1db/job.log
 (1/3) ras_tools.py:Ras_tools.test1_nvsetenv: PASS (0.27 s)
 (2/3) ras_tools.py:Ras_tools.test2_usysattn: PASS (0.28 s)
 (3/3) ras_tools.py:Ras_tools.test3_usysfault: PASS (0.28 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-08-04T16.42-1d0c1db/results.html
JOB TIME   : 1.15 s

Rhel
\# avocado run ras_tools.py 
JOB ID     : 9c62fbc32f51caa95a1d21d35cac47dae593ae38
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-08-04T16.44-9c62fbc/job.log
 (1/3) ras_tools.py:Ras_tools.test1_nvsetenv: PASS (0.86 s)
 (2/3) ras_tools.py:Ras_tools.test2_usysattn: PASS (0.82 s)
 (3/3) ras_tools.py:Ras_tools.test3_usysfault: PASS (0.74 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-08-04T16.44-9c62fbc/results.html
JOB TIME   : 3.03 s

File: ras/ras_tools.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+class Ras_tools(Test):
+    """
+    :avocado: tags=privileged
+    """
+    fail_cmd = list()
+
+    def run_cmd(self, cmd):
+        cmd_result = process.run(cmd, ignore_status=True, sudo=True,
+                                 shell=True)
+        if cmd_result.exit_status != 0:
```
Comment (PraveenPenguin) [inline ras/ras_tools.py:29]: may just use this way 

```
if process.system(cmd, ignore_status=True, sudo=True, shell=True):
   self.fail_cmd.append(cmd)
return
```

this way we can avoid using extra variable

File: ras/ras_tools.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.cancel("Fail to install %s required for this test." %
+                            pkg)
+
+    def test1_nvsetenv(self):
+        self.log.info("===Executing nvsetenv tool====")
+        self.is_fail = 0
+        self.run_cmd("nvsetenv")
+        self.run_cmd("nvsetenv load-base")
+        self.run_cmd("nvsetenv load-base 7000")
```
Comment (PraveenPenguin) [inline ras/ras_tools.py:52]: i suggest have these can be part of yml input and like we can have like  
```
value = self.params.get('nvsetenv_list', default=['load-base', 'load-base 7000'])
for list_item in value:
  self.run_cmd('nvsetenv  %s ' % list_item )
  ```

File: ras/ras_tools.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if self.is_fail >= 1:
```
Comment (PraveenPenguin) [inline ras/ras_tools.py:53]: `if self.is_fail:`  also result same

File: ras/ras_tools.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            self.fail("%s command(s) failed to execute  "
+                      % self.fail_cmd)
+
+    def test2_usysattn(self):
+        self.log.info("=====Executing usysattn tool test======")
+        self.is_fail = 0
+        self.run_cmd("usysattn -h")
+        self.run_cmd("usysattn -V")
```
Comment (PraveenPenguin) [inline ras/ras_tools.py:61]: can these also as yml input

File: ras/ras_tools.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        loc_code = self.run_cmd_out("usysattn -P| awk '{print $1}'")
+        self.run_cmd("usysattn -l %s -s normal -t" % loc_code)
+        if self.is_fail >= 1:
```
Comment (PraveenPenguin) [inline ras/ras_tools.py:64]: `if self.is_fail:` please

File: ras/ras_tools.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            self.fail("%s command(s) failed to execute  "
+                      % self.fail_cmd)
+
+    def test3_usysfault(self):
+        self.log.info("======Executing usysfault tool test======")
+        self.is_fail = 0
+        self.run_cmd("usysfault -h")
+        self.run_cmd("usysfault -V")
```
Comment (PraveenPenguin) [inline ras/ras_tools.py:72]: here also please yaml input

File: ras/ras_tools.py (hunk 1 above)
Comment (shirishaganta1) [reply ras/ras_tools.py:29]: sure will do that

File: ras/ras_tools.py (hunk 2 above)
Comment (shirishaganta1) [reply ras/ras_tools.py:52]: okay

File: ras/ras_tools.py (hunk 3 above)
Comment (shirishaganta1) [reply ras/ras_tools.py:53]: okay

File: ras/ras_tools.py (hunk 6 above)
Comment (shirishaganta1) [reply ras/ras_tools.py:72]: Addressed all the above comments and changed the script. Please check it once

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**